Sources/TSCBasic/Lock.swift

```diff
@@ -178,7 +178,23 @@ public final class FileLock {
             throw FileSystemError(.notDirectory, lockFilesDirectory)
         }
         // use the parent path to generate unique filename in temp
-        var lockFileName = (resolveSymlinks(fileToLock.parentDirectory).appending(component: fileToLock.basename)).components.joined(separator: "_") + ".lock"
+        var lockFileName = (resolveSymlinks(fileToLock.parentDirectory)
+                                .appending(component: fileToLock.basename))
+                                .components.joined(separator: "_")
+                                .replacingOccurrences(of: ":", with: "_") + ".lock"
+#if os(Windows)
+        // NTFS has an ARC limit of 255 codepoints
+        var lockFileUTF16 = lockFileName.utf16.suffix(255)
+        while String(lockFileUTF16) == nil {
+            lockFileUTF16 = lockFileUTF16.dropFirst()
+        }
+        lockFileName = String(lockFileUTF16) ?? lockFileName
+
+        let path = lockFilesDirectory.appending(component: lockFileName)
+        if path.pathString.utf8.count > Int(MAX_PATH) {
+            lockFileName = String(lockFileName.dropFirst(path.pathString.utf8.count - Int(MAX_PATH)))
```

Comment:
I don't think this will work since the count for `dropFirst()` is in characters but the count being checked is in UTF-8 bytes.

Reply:
I kind of feel that the PATH_MAX check can probably be deferred — the other code path doesn't check for that either.

Reply:
So just delete this second part for now, since that's generic anyway.

Reply:
Ah right - okay, let's make this entirely characters.  This should _mostly_ work right on Windows due to UTF16 (yes, surrogate pairs, blah, blah, blah, I said **mostly**).

Reply:
> So just delete this second part for now, since that's not actually platform specific anyway.

Sounds reasonable to me.  Better to be uniform.